thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/metric/transfer/MetricTransfer.java

```diff
@@ -0,0 +1,34 @@
+package com.linkedin.thirdeye.detector.metric.transfer;
+
+import com.linkedin.thirdeye.api.MetricTimeSeries;
+import java.util.List;
+
+
+public class MetricTransfer {
+
+  /**
+   * Use the scaling factor to normalize the input time series
+   *
+   * @param metricsToModify the data MetricTimeSeries to be modified by scaling factor
+   * @param metricName the metricName withing the timeseries to be modified
+   * @param scalingFactorList list of scaling factors
+   * @return
+   */
+  public static void rescaleMetric(MetricTimeSeries metricsToModify, String metricName,
+      List<ScalingFactor> scalingFactorList) {
+    if (scalingFactorList == null || scalingFactorList.size() < 1) {
+      return;  // no transformation if there is no scaling factor in
+    }
+    // Went over the List of Scaling Factors and Rescale the timestamps with in the time range
+    // get the metric name
+    for (long ts : metricsToModify.getTimeWindowSet()) {
```

Comment:
its better to invert the loop, if we know the time buckets for which we apply the scaling factor, we should go over that instead of going over the entire timeseries and checking if there is a scalingfactor. Also this assumes that scaling factor is represents the factor as the same granularity as data. For example, if scaling is 1.2 for nov 1 to nov 5. We have to break it down to 5 minute interval and set scaling factor 1.2 for every 5 minutes.  This is ok for now but we need to change to fix this.